package/lib/src/beam_location.dart

```diff
@@ -228,7 +228,16 @@ abstract class BeamLocation<T extends RouteInformationSerializable>
     } else {
       if (routeInformation == null) {
         updateState(history.last.routeInformation);
-      } else {
+      } else if (routeInformation.location == state.routeInformation.location) {
+        /// if the new route information is the same as in the state it means 
+        /// the state changed first and notified listeners, so updating it
+        /// will be unnecessary. Let's just add route to history with [tryPoppingHistory] set to true
+          addToHistory(
+          state.routeInformation,
+          beamParameters ?? const BeamParameters(),
+        );
+      } 
+      else {
```

Comment:
Only this change fixes the issue :muscle: (if I'm not mistaken)
I feel like we can fix it differently, but this is a good hotfix for now.
Thank you :slightly_smiling_face:

Reply:
yes, that's the actual fix 🙂